lang/java/reef-wake/wake/src/main/java/org/apache/reef/wake/remote/transport/netty/NettyChannelInitializer.java

```diff
@@ -19,34 +19,77 @@
 package org.apache.reef.wake.remote.transport.netty;
 
 import io.netty.channel.ChannelInitializer;
+import io.netty.channel.ChannelPipeline;
 import io.netty.channel.socket.SocketChannel;
 import io.netty.handler.codec.LengthFieldBasedFrameDecoder;
 import io.netty.handler.codec.LengthFieldPrepender;
 import io.netty.handler.codec.bytes.ByteArrayDecoder;
 import io.netty.handler.codec.bytes.ByteArrayEncoder;
+import io.netty.handler.codec.http.*;
+import org.apache.reef.wake.remote.transport.TransportFactory.ProtocolType;
+
 
 /**
  * Netty channel initializer for Transport.
  */
 class NettyChannelInitializer extends ChannelInitializer<SocketChannel> {
+
   /**
    * the buffer size of the frame decoder.
    */
   public static final int MAXFRAMELENGTH = 10 * 1024 * 1024;
+  private static final int MAX_HTTP_MESSAGE_LENGTH = 10 * 1024 * 1024;
   private final NettyChannelHandlerFactory handlerFactory;
 
-  NettyChannelInitializer(final NettyChannelHandlerFactory handlerFactory) {
+  /**
+   * Type of protocol channel use.
+   */
+  private final ProtocolType protocolType;
+  private final boolean isServer;
+
+  NettyChannelInitializer(
+      final NettyChannelHandlerFactory handlerFactory,
+      final ProtocolType protocol) {
+    this(handlerFactory, protocol, false);
+  }
+
+  NettyChannelInitializer(
+      final NettyChannelHandlerFactory handlerFactory,
+      final ProtocolType protocol,
+      final boolean isServer) {
     this.handlerFactory = handlerFactory;
+    this.protocolType = protocol;
+    this.isServer = isServer;
   }
 
   @Override
   protected void initChannel(final SocketChannel ch) throws Exception {
-    ch.pipeline()
-        .addLast("frameDecoder", new LengthFieldBasedFrameDecoder(MAXFRAMELENGTH, 0, 4, 0, 4))
-        .addLast("bytesDecoder", new ByteArrayDecoder())
-        .addLast("frameEncoder", new LengthFieldPrepender(4))
-        .addLast("bytesEncoder", new ByteArrayEncoder())
-        .addLast("chunker", new ChunkedReadWriteHandler())
-        .addLast("handler", handlerFactory.createChannelInboundHandler());
+    final ChannelPipeline pipeline = ch.pipeline();
+    switch (this.protocolType) {
+    case TCP:
+      pipeline
+          .addLast("frameDecoder", new LengthFieldBasedFrameDecoder(MAXFRAMELENGTH, 0, 4, 0, 4))
+          .addLast("bytesDecoder", new ByteArrayDecoder())
+          .addLast("frameEncoder", new LengthFieldPrepender(4))
+          .addLast("bytesEncoder", new ByteArrayEncoder())
+          .addLast("chunker", new ChunkedReadWriteHandler());
+      break;
+    case HTTP:
+      if(isServer) {
+        pipeline
+            .addLast("codec", new HttpServerCodec())
+            .addLast("requestDecoder", new HttpRequestDecoder())
```

Comment:
I followed example in the Netty Http server. But by reference document, I think we do need additional codec since `HttpServerCodec` is combination of them. I'll remove and test about it.